flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -4394,11 +4450,54 @@ class ArrayExprLowering {
         }(),
         x);
   }
+  static mlir::Type unwrapBoxEleTy(mlir::Type ty) {
+    if (auto boxTy = ty.dyn_cast<fir::BoxType>()) {
+      ty = boxTy.getEleTy();
+      if (auto refTy = fir::dyn_cast_ptrEleTy(ty))
+        ty = refTy;
+    }
+    return ty;
+  }
+  llvm::SmallVector<mlir::Value> getShape(mlir::Type ty) {
+    llvm::SmallVector<mlir::Value> result;
+    ty = unwrapBoxEleTy(ty);
+    auto loc = getLoc();
+    auto idxTy = builder.getIndexType();
+    for (auto extent : ty.cast<fir::SequenceType>().getShape()) {
+      auto v = extent == fir::SequenceType::getUnknownExtent()
+                   ? builder.create<fir::UndefOp>(loc, idxTy).getResult()
+                   : builder.createIntegerConstant(loc, idxTy, extent);
+      result.push_back(v);
+    }
+    return result;
+  }
+  llvm::SmallVector<mlir::Value>
+  getShape(const Fortran::semantics::SymbolRef &x) {
+    if (x.get().Rank() == 0)
+      return {};
+    return getFrontEndShape(x);
+  }
+  template <typename A>
+  llvm::SmallVector<mlir::Value> getShape(const A &x) {
+    if (x.Rank() == 0)
+      return {};
+    return getFrontEndShape(x);
+  }
+  template <typename A>
+  llvm::SmallVector<mlir::Value> getFrontEndShape(const A &x) {
+    if (auto optShape = Fortran::evaluate::GetShape(x)) {
+      llvm::SmallVector<mlir::Value> result;
+      convertFEShape(*optShape, result);
+      if (!result.empty())
+        return result;
```

Comment:
What is the purpose of this `if (!result.empty())` ? If result is empty, it looks like an empty vector is returned anyway.

Reply:
Thanks. It was left over from some other experimenting.